auth/resource.go

```diff
@@ -2,175 +2,149 @@ package auth
 	}
 
-	newResource := &Resource{
-		ResourceID:   resourceID,
-		PolicyID:     policyID,
-		PermissionID: permissionID,
-	}
+	log.Debug(ctx, map[string]interface{}{
+		"space_id":    spaceID,
+		"resource_id": resource.Data.ResourceID,
+	}, "Space resource created")
 
-	return newResource, nil
+	return resource, nil
 }
 
-func getPat(ctx context.Context, requestData *goa.RequestData, config KeycloakConfiguration) (string, error) {
-	endpoint, err := config.GetKeycloakEndpointToken(requestData)
-	if err != nil {
-		return "", err
-	}
-	token, err := GetProtectedAPIToken(ctx, endpoint, config.GetKeycloakClientID(), config.GetKeycloakSecret())
-	if err != nil {
-		return "", err
+// DeleteSpace calls auth service to delete the keycloak resource associated with the space
+func (m *AuthzResourceManager) DeleteSpace(ctx context.Context, request *goa.RequestData, spaceID string) error {
+	if !m.configuration.IsAuthorizationEnabled() {
+		// Keycloak authorization is disabled by default in Developer Mode
+		return nil
 	}
-	return token, nil
-}
-
-// DeleteResource deletes the keycloak resource and associated permission and policy
-func (m *KeycloakResourceManager) DeleteResource(ctx context.Context, request *goa.RequestData, resource Resource) error {
-	authzEndpoint, err := m.configuration.GetKeycloakEndpointAuthzResourceset(request)
+	c, err := m.createClient(ctx, request)
 	if err != nil {
 		return err
 	}
-	clientsEndpoint, err := m.configuration.GetKeycloakEndpointClients(request)
+	sUD, err := goauuid.FromString(spaceID)
 	if err != nil {
 		return err
 	}
-	pat, err := getPat(ctx, request, m.configuration)
+	res, err := c.DeleteSpace(goasupport.ForwardContextRequestID(ctx), authservice.CreateSpacePath(sUD))
 	if err != nil {
-		return err
+		log.Error(ctx, map[string]interface{}{
+			"space_id": spaceID,
+			"err":      err.Error(),
+		}, "unable to delete a space resource via auth service")
+		return errs.Wrap(err, "unable to delete a space resource via auth service")
 	}
-	publicClientID := m.configuration.GetKeycloakClientID()
-	clientID, err := GetClientID(context.Background(), clientsEndpoint, publicClientID, pat)
-	if err != nil {
-		return err
+	defer res.Body.Close()
+	if res.StatusCode != http.StatusOK {
+		log.Error(ctx, map[string]interface{}{
+			"space_id":        spaceID,
+			"response_status": res.Status,
+			"response_body":   rest.ReadBody(res.Body),
+		}, "unable to delete a space resource via auth service")
+		return errs.Errorf("unable to delete a space resource via auth service. Response status: %s. Responce body: %s", res.Status, rest.ReadBody(res.Body))
 	}
 
-	// Delete resource
-	err = DeleteResource(ctx, resource.ResourceID, authzEndpoint, pat)
+	log.Debug(ctx, map[string]interface{}{
+		"space_id": spaceID,
+	}, "Space resource deleted")
+
+	return nil
+}
+
+func (m *AuthzResourceManager) createClient(ctx context.Context, request *goa.RequestData) (*authservice.Client, error) {
```

Comment:
To avoid repetition of this block of code when we create clients for user profile service, login,etc, we should move this out to somewhere more generic.

Reply:
I agree but we can move it to something else in the next PR when we start use it in other places.